plugins/PostProcessingPlugin/scripts/SetInitialSwitchedExtruderOffset.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) 2022 Anson Liu
+# This script is released under the terms of the AGPLv3 or higher.
+
+from ..Script import Script
+from UM.Logger import Logger
+from UM.Message import Message
+from cura.CuraApplication import CuraApplication
+import re
+
+class ExtruderInitialOffsetInfo:
+  def __init__(self, offset_complete, switch_extruder_retraction_amount, switch_extruder_prime_speed, switch_extruder_extra_prime_amount, retraction_distance):
+    self.offset_complete = offset_complete
+    self.switch_extruder_retraction_amount = switch_extruder_retraction_amount
+    self.switch_extruder_prime_speed = switch_extruder_prime_speed
+    self.switch_extruder_extra_prime_amount = switch_extruder_extra_prime_amount
+    self.retraction_distance = retraction_distance
+
+class SetInitialSwitchedExtruderOffset(Script):
+    """Offsets extruders by Nozzle Switch Retraction Distance the first time they are using in a print. 
+
+    This primes filament flow on extruders that were switched away from and had filament retracted a longer distance out of the melt zone.
+    """
+    def __init__(self):
+        super().__init__()
+
+    def getSettingDataString(self):
+        return """{
+            "name": "Set Initial Switched Extruder Offset",
+            "key": "SetInitialSwitchedExtruderOffset",
+            "metadata": {},
+            "version": 2,
+            "settings":
+            {
+                "addSwitchExtruderExtraPrimeAmount":
+                {
+                    "label": "Add Nozzle Switch Extra Prime Amount to Offset",
+                    "description": "Offset the initial offset of the extruder by an additional length equal to that extruder's Nozzle Switch Extra Prime Amount.",
+                    "type": "bool",
+                    "default_value": true
+                },
+                "setT0ExtruderOffset":
+                {
+                    "label": "Set Extruder 1 Offset",
+                    "description": "Set the initial offset of this extruder by Nozzle Switch Retraction Distance. If this extruder was previously switched out and retracted, this will prime filament flow the first time this extruder is using this print. If your machine primes the initial extruder, the initial extruder does not need an offset.",
+                    "type": "bool",
+                    "default_value": false
+                },
+                "setT1ExtruderOffset":
+                {
+                    "label": "Set Extruder 2 Offset",
+                    "description": "Set the initial offset of this extruder by Nozzle Switch Retraction Distance.",
+                    "type": "bool",
+                    "default_value": "CuraApplication.getInstance().getExtruderManager().getInitialExtruderNr() != 1"
+                },
+                "setT2ExtruderOffset":
```

Comment:
how about adding a parameter "enabled" to see if the extruder exist in the printer and then show the parameter to set the offset of extruder?

Reply:
Which Cura variable should I use to get the "enabled" state of an extruder index? I had tried to get the extruder enabled state previously but got an error trying to access the value.

Reply:
try "machine_extruder_count"

Reply:
I tried reading the value during getSettingDataString() in the "enabled" json field like below but it always evaluates to False

```json
"setT1ExtruderOffset":
                {
                    "label": "Set Extruder 2 Offset",
                    "description": "Set the initial offset of this extruder by Nozzle Switch Retraction Distance.",
                    "type": "bool",
                    "default_value": "CuraApplication.getInstance().getExtruderManager().getInitialExtruderNr() != 1",
                    "enabled": "CuraApplication.getInstance().getGlobalContainerStack().getProperty(\\\"machine_extruder_count\\\", \\\"value\\\") >= 2"
                },
```

When I read the value normally right before the json string in getSettingDataString() it logs the correct values in cura.log

```python
logthis = f'SetInitialSwitchedExtruderOffset machine_extruder_count {CuraApplication.getInstance().getGlobalContainerStack().getProperty("machine_extruder_count", "value")}'
log2= f'machine_extruder_count>=2 {CuraApplication.getInstance().getGlobalContainerStack().getProperty("machine_extruder_count", "value") >= 2}'
Logger.log("d", logthis)
Logger.log("d", log2)
```

I looked at the other plugins to find any "enabled" examples that call python and they only reference other settings fields. Is there a restriction on running Python in "enabled"?

Reply:
From the Cura logs, it seems like "enabled" code is unable to call blacklisted methods that are only blacklisted for the json?
```
2024-04-24 23:41:39,953 - ERROR - [MainThread] UM.Settings.SettingFunction._safeCompile [77]: Use of illegal method extruderIds in function ("2" in CuraApplication.getInstance().getExtruderManager().extruderIds) for setting
2024-04-24 23:41:39,953 - ERROR - [MainThread] UM.Settings.SettingFunction._safeCompile [77]: Use of illegal method getProperty in function (CuraApplication.getInstance().getGlobalContainerStack().getProperty("machine_extruder_count", "value") >= 4) for setting
```